src/pages/moderators-page/components/main-content/index.tsx

```diff
@@ -1,30 +1,136 @@
+	ReviewModeratorsCard,
+} from "./components/index";
 import styles from "./styles.module.scss";
 
+const INDEX_ZERO = 0;
+
 const MainModeratorsContent: React.FC = () => {
-	const { data: moderatorsReviews } = useGetReviewsModerationQuery();
+	const [searchTerm, setSearchTerm] = useState<string>("");
+	const [filterByStatus, setFilterByStatus] =
+		useState<DropdownOption["value"]>();
+	const [sortByPeriod, setSortByPeriod] = useState<DropdownOption["value"]>();
+	const [filterByType, setFilterByType] =
+		useState<keyof typeof ButtonGroupData>("Компанії");
+	const [isOpenSerchFiltersModal, setIsOpenSerchFiltersModal] = useState(false);
+	const screenWidth = useGetScreenWidth();
+
+	const handleSetSearchTerm = (term: string) => {
+		setSearchTerm(term);
+	};
+
+	// const handleOpenSearchFilter = () => {
+	// 	setIsOpenSerchFiltersModal(true);
+	// };
+
+	const handleSetFilterByStatus = (sortOption: DropdownOption["value"]) => {
+		if (sortOption) setFilterByStatus(sortOption);
+		if (!sortOption) setFilterByStatus(undefined);
+	};
+	const handleSetSortByPeriod = (sortOption: DropdownOption["value"]) => {
+		if (sortOption) setSortByPeriod(sortOption);
+		if (!sortOption) setSortByPeriod(undefined);
+	};
+
+	const fetchResult = useReviewModerationApi({
+		id: searchTerm,
+		ordering: sortByPeriod as GetModerationReviewsRequest["ordering"],
+		status: filterByStatus as GetModerationReviewsRequest["status"],
+		type: ButtonGroupData[filterByType],
+	});
+	// const err = (fetchResult?.error as FetchBaseQueryError)?.status;
+	// const loadError = (error as FetchBaseQueryError)?.data
+	// 	? ((error as FetchBaseQueryError).data as Error)
+	// 	: { message: "Невідома помилка" };
+	// setServerError(loadError.message);
+
 	const dispatch = useDispatch();
 
 	useEffect(() => {
-		if (moderatorsReviews) {
-			dispatch(setRewiews(moderatorsReviews.results));
+		if (fetchResult?.reviews) {
+			dispatch(setRewiews(fetchResult?.reviews.results));
 		}
-	}, [moderatorsReviews, dispatch]);
+	}, [fetchResult?.reviews, dispatch]);
 
 	return (
 		<div className={styles["moderators_wrapper"]}>
-			<div className={styles["title_wrapper"]}>
-				<p>Модерація відгуків</p>
-				<p>Знайдено: {moderatorsReviews?.count} відгуків</p>
-			</div>
-			<div>
-				<p>Пошук за UID</p>
-			</div>
+			<header className={styles["header_wrapper"]}>
+				<h2 className={styles["title"]}>Модерація відгуків</h2>
+				<p className={styles["sub_title"]}>
+					Знайдено:{" "}
+					<span>
+						{fetchResult?.error ? INDEX_ZERO : fetchResult?.reviews?.count}
+					</span>{" "}
+					відгуків
+				</p>
+			</header>
+
+			<section className={styles["moderators_fitters_section"]}>
+				<div className={styles["search_block"]}>
+					<p className={styles["search_title"]}>Пошук за UID</p>
+					<SearchBar
+						filterButtonText=" "
+						filtersLength={3}
+						isFilterButton={screenWidth <= ScreenBreakpoints.MOBILE}
+						onOpenFilter={() => setIsOpenSerchFiltersModal((prev) => !prev)}
```

Comment:
create handler for this

Reply:
you think it wiil be better to create more code?

Reply:
++